src/formatter.rs

```diff
@@ -92,17 +92,29 @@ impl Formatter {
         result
     }
 
+    pub fn format_testcase(&self, testcase: &ClashTestCase, style: &OutputStyle, header: String) -> String {
```

Comment:
Taking a `String` parameter when you are only reading it is bad because it forces the caller to clone a borrowed string if they want to pass it to the function (like you did in clash.rs:117 with `.to_string()`). You should take `header: &str` instead, that way the caller has more flexibility.

Here's a short article that probably does a better job at explaining it than I did: http://xion.io/post/code/rust-string-args.html